### PR TITLE
support toJSON with embedded dataclass

### PR DESCRIPTION
**Pull request by mingchuno**
fix currently dataclass with embedded dataclass cannot be correctly
serialize to JSON. Also dataclass with non-primitive value cannot be
correctly compare by equals which have no fix at the moment

### CODE REVIEW

**Comment by mingchuno** [comment]
@alexeyraspopov

File: modules/Record.js [hunk 1]
```diff
@@ -54,7 +54,8 @@ export default class Record {
     let result = {};
 
     for (let key in this.constructor[defaults]) {
-      result[key] = this[key];
+      let value = this[key]
+      result[key] = (value && value.toJSON) ? value.toJSON() : value
```
Comment (alexeyraspopov) [inline modules/Record.js]: Let's test for `typeof value.toJSON === 'function'`, to make sure that the call is possible.

Comment (alexeyraspopov) [reply modules/Record.js]: Also, don't forget semicolons :)

File: modules/__tests__/Record-test.js [hunk 2]
```diff
@@ -118,6 +118,46 @@ describe('Record', () => {
...
+      },
+      date: "1996-12-16T19:24:00.000Z",
+      obj: {
+        foo: "bar"
+      }
+    };
+    expect(JSON.stringify(embedded)).toBe(JSON.stringify(raw));
+  })
+
+  // FIXME: equals cannot compare dataclass with value other than primitive value
```
Comment (alexeyraspopov) [inline modules/__tests__/Record-test.js:150]: Thanks for putting this test here. I have an idea for fixing it, can be done after merging the PR.

**Review by alexeyraspopov — COMMENTED** [review]
Sorry for taking so long for review. Was busy with daily work.

There is one small thing I'd ask you to update. Overall changes looks great. Thanks for your contribution.

**Comment by mingchuno** [comment]
@alexeyraspopov Hi, I have done the update, please check

**Pull request merged**